_gsocblogs/2024/blog_SOFIE_VedantMehra.md

```diff
@@ -0,0 +1,220 @@
+</div> 
+
+
+## Details
+
+|  |  |
+| --- | --- |
+| Name | [Vedant Mehra](https://github.com/extint) |
+| Organisation | [CERN HSF (Root Project)](https://github.com/root-project/root) |
+| Mentor | [Dr. Lorenzo Moneta](https://github.com/lmoneta), [Sanjiban Sengupta](https://github.com/sanjibansg), [Neel Shah](https://github.com/neel-shah-29)|
+| Project | [ROOT - TMVA SOFIE Developments - Inference Code Generation for Deep Learning models](https://summerofcode.withgoogle.com/programs/2024/projects/6DDnly0z) |
+
+
+## Table of Contents
+
+1. [Project Description](#project-description)
+2. [SOFIE: System for Optimized Fast Inference code Emit](#sofie-system-for-optimized-fast-inference-code-emit)
+3. [GSoC 2024 Objectives](#gsoc-2022-objectives)
+4. [Work Accomplished](#work-accomplished)
+   - [New Operator Implementations](#1-implementing-New-Operator-Classes)
+   - [Extending Existing Operators](#2-adapting-existing-operator-classes-to-parse-parameterized-shapes)
+5. [Challenges and Learning Outcomes](#challenges-and-learning-outcomes)
+6. [Future Work](#Future-Work)
+7. [Conclusion](#conclusion)
+
+## Project Description
+
+The Toolkit for Multivariate Analysis (TMVA) is a part of the ROOT Project, providing a machine learning framework used primarily in High-Energy Physics for tasks like multivariate classification and regression. Given the massive data output (often reaching petabytes daily) from research activities, and the need for fast, real-time predictions, a highly efficient inference engine is necessary for using trained machine learning models regularly. With this objective SOFIE (System for Optimized Fast Inference code Emit) was developed.
+
+### SOFIE (System for Optimized Fast Inference code Emit)
+ROOT/TMVA SOFIE generates C++ functions easily invokable for the fast inference of trained neural network models. It takes ONNX model files as inputs and produces C++ header files that can be included and utilized in a “plug-and-go” style.
+
+```c
+// This generates a .hxx file containing the C++ code for model inference.
+using namespace TMVA::Experimental;
+SOFIE::RModelParser_ONNX Parser;
+SOFIE::RModel model = Parser.Parse("./example_model.onnx");
+model.Generate();
+model.OutputGenerated("./example_output.hxx");
+```
+
+SOFIE uses BLAS or Eigen for executing the inference and Protobuf3 to parse ONNX files. It features two main classes: ```RModel``` for holding model data and ```ROperator``` for handling ONNX operations and generating inference code. Another sub-directory, `sofie-parsers`, was added under TMVA to include the ONNX parser, RModelParser_ONNX. This parser reads ONNX files, loads their values into an RModel object, and returns it. It creates an intermediate representation of the model and outputs a C++ header file with an infer function that can be directly used in production environments.
+
+![flowchart_sofie](https://raw.githubusercontent.com/extint/gsoc24-blog/main/assets/Blank%20board%20-%20Page%201.png)
```

Comment:
```suggestion
<img src="https://raw.githubusercontent.com/extint/gsoc24-blog/main/assets/Blank%20board%20-%20Page%201.png" alt="Flowchart Sophie" width="100%" >
```
Otherwise the image seems to render too big